rpc/src/v1/impls/eth.rs

```diff
@@ -672,18 +672,22 @@ impl<C, M> EthFilter for EthFilterClient<C, M> where
 							to_value(&hashes)
 						},
 						PollFilter::PendingTransaction(ref mut previous_hashes) => {
+							// get hashes of pending transactions
 							let current_hashes = take_weak!(self.miner).pending_transactions_hashes();
-							// calculate diff
 							let previous_hashes_set = previous_hashes.into_iter().map(|h| h.clone()).collect::<HashSet<H256>>();
```

Comment:
`pending_transactions_hashes()` returns a Vec<H256> -- why is the clone necessary? You already own the data.

```rust
use std::iter::FromIterator;

let previous_hashes_set = HashSet::from_iterator(take_weak!(self.miner).pending_transactions_hashes());
```